runconfig/hostconfig.go

```diff
@@ -176,6 +177,7 @@ func ContainerHostConfigFromJob(job *engine.Job) *HostConfig {
 		MemorySwap:      job.GetenvInt64("MemorySwap"),
 		CpuShares:       job.GetenvInt64("CpuShares"),
 		CpusetCpus:      job.Getenv("CpusetCpus"),
+		OomKillDisable:  job.GetenvBool("OomKillDisable"),
```

Comment:
We should not add it here, all these codes below `if job.EnvExists("HostConfig")` is for backward compatibility, the new comings should always in `HostConfig` in json, or it won't work. We should lead users to use it the right way.